src/firmware/app/control/trajectory_planner_test.cpp

```diff
@@ -1288,50 +768,1631 @@ TEST_F(TrajectoryPlannerTest, velocity_trajectory_curved_path_low_acceleration)
+    path_parameters.t_start = 0;
+    path_parameters.t_end   = 1;
+    path_parameters.num_segments =
+        TRAJECTORY_PLANNER_MAX_NUM_ELEMENTS;  // For tests the number of segments is
+    // arbitrary, so use the maximum possible
+    path_parameters.max_allowable_linear_acceleration = 3;
+    path_parameters.max_allowable_linear_speed        = 3;
+    path_parameters.initial_linear_speed              = 0;
+    path_parameters.final_linear_speed                = 0;
+
+    path_parameters.orientation_profile                = orientation_profile;
+    path_parameters.max_allowable_angular_acceleration = 9;
+    path_parameters.max_allowable_angular_speed        = 6;
+
+
+    PositionTrajectoryElement_t const_t_segments[TRAJECTORY_PLANNER_MAX_NUM_ELEMENTS];
+    TrajectorySegment_t segment_lengths[TRAJECTORY_PLANNER_MAX_NUM_ELEMENTS];
+    PositionTrajectory_t trajectory = {.trajectory_elements = const_t_segments,
+                                       .path_parameters     = path_parameters};
+
+    app_trajectory_planner_generateStatesAndReturnSegmentLengths(&trajectory,
+                                                                 segment_lengths);
+
+    const float delta_length = total_path_length_linear / path_parameters.num_segments;
+    const float delta_length_orientation =
+        total_path_length_angular / path_parameters.num_segments;
+    // Check that all of the segment lengths are correct
+    for (unsigned int i = 0; i < path_parameters.num_segments - 1; i++)
+    {
+        EXPECT_NEAR(segment_lengths[i].linear_segment_length, delta_length, 0.00001);
+        EXPECT_NEAR(segment_lengths[i].angular_segment_length, delta_length_orientation,
+                    0.00001);
+    }
+}
+
+TEST_F(TrajectoryPlannerTest,
+       test_robot_segment_size_generation_at_constant_t_intervals_low_segment_count)
+{
+    const float total_path_length_linear  = sqrt(2.0);
+    const float total_path_length_angular = 1;
+    Polynomial2dOrder3_t path             = {
+        .x = {.coefficients = {0, 0, 1, 0}},
+        .y = {.coefficients = {0, 0, 1, 0}},
+
+    };
+
+    Polynomial1dOrder3_t orientation_profile = {.coefficients = {0, 0, 1, 0}};
+    // Generate parameters for a trajectory
+    FirmwareRobotPathParameters_t path_parameters;
+    path_parameters.path         = path;
+    path_parameters.t_start      = 0;
+    path_parameters.t_end        = 1;
+    path_parameters.num_segments = 5;  // For tests the number of segments is
+    // arbitrary, so use the maximum possible
+    path_parameters.max_allowable_linear_acceleration = 3;
+    path_parameters.max_allowable_linear_speed        = 3;
+    path_parameters.initial_linear_speed              = 0;
+    path_parameters.final_linear_speed                = 0;
+
+    path_parameters.orientation_profile                = orientation_profile;
+    path_parameters.max_allowable_angular_acceleration = 9;
+    path_parameters.max_allowable_angular_speed        = 6;
```

Comment:
Nit; but we should prefer struct initialisation, ie:
```
const FirmwareRobotPathParameters path_parameters {
   .path = path,
   .t_start = 0,
...
};
```
Same goes for the other cases of this in tests. This helps insure that the object is always valid, and has the added benefit of being a bit neater!

Reply:
Redone all path param struct initialization that use all the variables.

For tests that don't I only initialize the information needed to simplify.